src/App/Repository/UserRepository.php

```diff
@@ -108,6 +110,39 @@ public function createWithHash(Entity $entity)
         return parent::create($entity);
     }
 
+    public function createMany(Collection $collection) : array
+    {
+        // Check MySQL `innodb_autoinc_lock_mode` = 0 or 1 before running
+        $lockMode = DB::query(Database::SELECT, "SHOW VARIABLES LIKE 'innodb_autoinc_lock_mode'")
+            ->execute($this->db())
+            ->get('Value');
+
+        if (!in_array((int) $lockMode, [0, 1])) {
+            throw new \RuntimeException('Cannot bulk insert users with innodb_autoinc_lock_mode = ' . $lockMode);
+        }
+
+        $first = $this->removeNullValues($collection->first()->asArray());
+        unset($first['contacts']);
+        $columns = array_keys($first);
+
+        $values = $collection->map(function ($entity) {
+            $data = $this->removeNullValues($entity->asArray());
```

Comment:
is that possible? I cant recall if we had optional stuff that maybe would break this. IDN

Reply:
Yea. You can pretty much leave any property empty, and because the function just takes a collection of entities it doesn't know anything. It's mainly a problem because `removeNullValues` might leave us with 2 values, or maybe 6. But we don't really know. 
The alternative was to skip it and try and insert nulls everywhere and have that fail.

Hopefully if it's going to break it breaks while I'm working on importing tags, posts, etc.

Reply:
Made it work without `removeNullValues` so it's a little more resilient (or at least fails predictably if you omit required values)